src/test/java/uk/gov/laa/gpfd/service/MappingTableServiceTest.java

```diff
@@ -0,0 +1,197 @@
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void shouldReturnReportListEntriesWhenValidReportsExist() {
+        // Given
+        var mappingTable1 = MappingTableTestDataFactory.aValidBankAccountReport();
+        var mappingTable2 = MappingTableTestDataFactory.aValidInvoiceAnalysisReport();
+        var list = Arrays.asList(mappingTable1, mappingTable2);
+        var expected = list.stream().map(ReportsGet200ResponseReportListInnerMapper::map).toList();
+
+        when(mappingTableDao.fetchReportList()).thenReturn(list);
+
+        // When
+        var result = mappingTableService.fetchReportListEntries();
+
+        // Then
+        assertNotNull(result);
+        assertEquals(expected.size(), result.size());
+        assertEquals(expected, result);
+        verify(mappingTableDao, times(1)).fetchReportList();
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenNoReportsExist() {
+        // Given
+        when(mappingTableDao.fetchReportList()).thenReturn(Collections.emptyList());
+
+        // When
+        var result = mappingTableService.fetchReportListEntries();
+
+        // Then
+        assertNotNull(result);
+        assertTrue(result.isEmpty());
+        verify(mappingTableDao, times(1)).fetchReportList();
+    }
+
+    @Test
+    void shouldReturnSingleReportWhenOnlyOneReportExists() throws DatabaseReadException {
+        // Given
+        var singleReport = MappingTableTestDataFactory.aValidInvoiceAnalysisReport();
+        when(mappingTableDao.fetchReportList()).thenReturn(Collections.singletonList(singleReport));
+        var expected = List.of(ReportsGet200ResponseReportListInnerMapper.map(singleReport));
+
+        // When
+        var result = mappingTableService.fetchReportListEntries();
+
+        // Then
+        assertNotNull(result);
+        assertEquals(expected.size(), result.size());
+        assertEquals(expected, result);
+        verify(mappingTableDao, times(1)).fetchReportList();
+    }
+
+    @Test
+    void shouldHandleSizeLimitExceededAndReturnAllReports() throws DatabaseReadException {
+        // Given
+        var largeList = new ArrayList<MappingTable>();
+        for (int i = 0; i < 1001; i++) {  // Exceed the size limit
+            largeList.add(MappingTableTestDataFactory.aValidInvoiceAnalysisReport());
+        }
+        when(mappingTableDao.fetchReportList()).thenReturn(largeList);
+
+        // When
+        var result = mappingTableService.fetchReportListEntries();
+
+        // Then
+        assertNotNull(result);
+        assertEquals(largeList.size(), result.size());
+        verify(mappingTableDao, times(1)).fetchReportList();
+    }
+
+    @Test
+    void shouldReturnValidReportsAndIgnoreInvalidData() throws DatabaseReadException {
+        // Given
+        var mappingTable1 = MappingTableTestDataFactory.aValidInvoiceAnalysisReport();
+        var mappingTable2 = new MappingTable(999, null, null, null, 0, null, null, null, null, null, null); // Invalid data
```

Comment:
Have we confirmed null values are invalid?

Reply:
Thank you for the suggestion! I agree that "WhenRequestIdIsZero" is a clear and descriptive name. However, I opted to go with the negative value scenario here to maintain consistency with the handling logic further down.

Reply:
I wanted to explicitly express the data that we might receive from the database to better align with and accurately map the model.